assets/js/components/surveys/SurveyQuestionMultiSelect.js

```diff
@@ -0,0 +1,199 @@
+ */
+import PropTypes from 'prop-types';
+import keyBy from 'lodash/keyBy';
+
+/**
+ * WordPress dependencies
+ */
+import { useState } from '@wordpress/element';
+
+/**
+ * Internal dependencies
+ */
+import Button from '../Button';
+import Checkbox from '../Checkbox';
+import SurveyHeader from './SurveyHeader';
+import { TextField, Input } from '../../material-components';
+
+const MAXIMUM_CHARACTER_LIMIT = 100;
+
+/* eslint-disable camelcase */
+
+const SurveyQuestionMultiSelect = ( {
+	question,
+	choices,
+	answerQuestion,
+	dismissSurvey,
+	min_choices = 1,
+	max_choices,
+} ) => {
+	const mappedChoices = choices.map( ( { answer_ordinal, write_in } ) => {
+		const optionalKeys = write_in ? { answer_text: '' } : {};
+
+		return {
+			answer_ordinal,
+			selected: false,
+			...optionalKeys,
+		};
+	} );
+
+	const initialState = keyBy( mappedChoices, 'answer_ordinal' );
+
+	const [ selectedValues, setSelectedValues ] = useState( initialState );
+
+	const handleCheck = ( answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				selected: ! selectedValues[ answer_ordinal ].selected,
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleAnswerChange = ( event, answer_ordinal ) => {
+		const newState = {
+			...selectedValues,
+			[ answer_ordinal ]: {
+				...selectedValues[ answer_ordinal ],
+				answer_text: event.target.value?.slice(
+					0,
+					MAXIMUM_CHARACTER_LIMIT
+				),
+			},
+		};
+
+		setSelectedValues( newState );
+	};
+
+	const handleSubmit = () => {
+		const answer = Object.values( selectedValues )
+			.filter( ( { selected } ) => selected )
+			.map( ( { answer_ordinal, answer_text } ) => {
+				if ( answer_text ) {
+					return { answer_ordinal, answer_text };
+				}
+				return { answer_ordinal };
+			} );
+		answerQuestion( {
+			answer,
+		} );
+	};
+
+	const hasEmptySelectedTextValue =
+		choices.filter( ( { write_in, answer_ordinal } ) => {
+			if ( write_in ) {
+				const { selected, answer_text } = selectedValues[
+					answer_ordinal
+				];
+				if ( selected && answer_text.length === 0 ) {
+					return true;
+				}
+			}
+			return false;
+		} ).length > 0;
+
+	const totalSelectedValues = Object.values( selectedValues ).filter(
+		( { selected } ) => selected
+	).length;
+	const hasLessThanMinChoices = totalSelectedValues < min_choices;
+	const hasMoreThanMaxChoices =
+		max_choices && totalSelectedValues > max_choices;
+
+	const isSubmitButtonDisabled =
+		hasEmptySelectedTextValue ||
+		hasLessThanMinChoices ||
+		hasMoreThanMaxChoices;
+
+	return (
+		<div className="googlesitekit-survey__multi-select">
+			<SurveyHeader title={ question } dismissSurvey={ dismissSurvey } />
+
+			<div className="googlesitekit-survey__body">
+				{ choices.map( ( { answer_ordinal, text, write_in } ) => {
+					const answer = selectedValues[ answer_ordinal ];
+
+					return (
+						<div key={ text } style={ { display: 'flex' } }>
```

Comment:
Let's not have style props set on elements unless we 100% have to 😄 

Can we assign a `className` to this instead?

Reply:
Ooops. I left that in from some rapid development :see_no_evil: 

FIxing now

Reply:
I'm also going to match the styling of the single select more closely

@asvinb  wrote the IB for this and I like the styling he chose for the text input (we have no designs for any of this!)

**single select**

![single select](https://user-images.githubusercontent.com/8496063/128855826-0e493994-cb78-48a6-8622-659ce8bb1509.png)


**multi select now** _will copy above_

![multiselect](https://user-images.githubusercontent.com/8496063/128855839-bdffb734-c5d7-44eb-a913-4d39e07ad41e.png)